EightBall.java

```diff
@@ -0,0 +1,17 @@
+package com.fortify.sca.plugins.maven.samples;
+
+import java.io.FileReader;
+
+public class EightBall {
+	public static void main(String args[]) throws Exception {
+		char[] buffer = new char[1024];
+		String filename = args[0];
+		try {
+			filename = "" + (Integer.parseInt(filename) % 3);
+		} catch (Exception e) {
+			System.out.println("Invalid input.");
```

Comment:
<p><b>Security Scanning</b> / Fortify SAST</p>
<h3>Low - Path Manipulation </h3>
<p>Attackers can control the file system path argument to FileReader() at EightBall.java line 12, which allows them to access or modify otherwise protected files.


</p>
<br>
<p><a href=http://localhost:2320/ssc/html/ssc/version/46/fix/7247/?engineType=SCA&issue=6291C6A33303ED270C269917AA8A1005&filterSet=f2f88863-0d8f-4d22-b001-3e1cd3ec0f45 target="_blank" rel="noopener noreferrer">More detailed information</a></p>